src/anim/evaluator/anim-clip.js

```diff
@@ -83,13 +89,52 @@ class AnimClip {
         this._blendOrder = blendOrder;
     }
 
+    get eventCursor() {
+        return this._eventCursor;
+    }
+
+    set eventCursor(value) {
+        this._eventCursor = value;
+    }
+
+    activeEventsForFrame(frameStartTime, frameEndTime) {
+        if (frameStartTime === 0) {
+            this.eventCursor = 0;
+        }
+        var events = [];
```

Comment:
instead of this function allocating an array each time so that the caller call fire events, it should fire them directly and avoid the allocations.

Reply:
alternatively, have a global array that you reuse for this reason